src/app/reporting/ReportSchedulerImpl.h

```diff
@@ -0,0 +1,63 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/reporting/ReportScheduler.h>
+
+namespace chip {
+namespace app {
+
+namespace reporting {
+
+class ReportSchedulerImpl : public ReportScheduler
+{
+public:
+    ReportSchedulerImpl(TimerDelegate * aTimerDelegate);
+    virtual ~ReportSchedulerImpl() override { UnregisterAllHandlers(); }
+
+    // ReadHandlerObserver
+    virtual void OnReadHandlerAdded(ReadHandler * aReadHandler) override;
```

Comment:
Please drop all the "virtual" in the places where you have "override".  "override" already implies "virtual".